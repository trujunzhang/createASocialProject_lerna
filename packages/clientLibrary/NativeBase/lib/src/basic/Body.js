import React, { Component } from "react";
import * as PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Body extends Component {
  render() {
    return <View ref={c => (this._root = c)} {...this.props} />;
  }
}

Body.propTypes = {
  ...ViewPropTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ])
};

const StyledBody = connectStyle("NativeBase.Body", {}, mapPropsToStyleNames)(
  Body
);

export { StyledBody as Body };
