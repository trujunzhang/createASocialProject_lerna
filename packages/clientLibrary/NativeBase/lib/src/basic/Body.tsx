import * as React from 'react'
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Body extends React.Component {
  private _root: any
  render() {
    return <View ref={c => (this._root = c)} {...this.props} />;
  }
}

// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBody = connectStyle("NativeBase.Body", {}, mapPropsToStyleNames)(
  Body
);

export { StyledBody as Body };
