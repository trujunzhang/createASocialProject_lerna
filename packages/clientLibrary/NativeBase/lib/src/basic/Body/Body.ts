import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface BodyProps {
  style?: RnViewStyleProp | Array<RnViewStyleProp>
}
