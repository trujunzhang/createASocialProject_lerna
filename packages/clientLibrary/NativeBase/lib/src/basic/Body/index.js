import * as React from 'react';
import { View } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { mapPropsToStyleNames } from '@app/tools';
class Body extends React.Component {
    render() {
        return React.createElement(View, Object.assign({ name: 'body', ref: (c) => (this._root = c) }, this.props));
    }
}
// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }
const StyledBody = connectStyle('NativeBase.Body', {}, mapPropsToStyleNames)(Body);
export { StyledBody as Body };
//# sourceMappingURL=index.js.map