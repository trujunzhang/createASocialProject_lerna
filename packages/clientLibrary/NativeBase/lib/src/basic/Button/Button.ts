import * as React from 'react'

import { ReactNative, RnViewStyleProp, RnTextStyleProp, BsStyle } from '@appUtils/rnCommonProps'

/**
 * see Widget Button.js
 */
export interface ButtonProps extends ReactNative.TouchableOpacityProps, BsStyle {
  /**
   * Defines button style
   */
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  /**
   * Defines button text style
   */
  textStyle?: RnTextStyleProp
  // onPress?: () => void
  /**
   * Block level button
   */
  block?: boolean
  /**
   * Vertical button
   */
  vertical?: boolean
  /**
   * Button with badges
   */
  badge?: boolean
  //primary?: boolean,
  /**
   * Gives you effect of Icon-buttons.
   * To have button with transparent background, include this prop.
   */
  transparent?: boolean
  //success?: boolean,
  //danger?: boolean,
  // warning?: boolean,
  //info?: boolean,
  color?: string
  /**
   * Applies outline button style.
   */
  bordered?: boolean
  /**
   * Renders button with slightly round shaped edges.
   */
  rounded?: boolean
  /**
   * For large size button
   */
  large?: boolean
  /**
   * For small size button
   */
  small?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the left in button.
   * By default, icons are aligned to the left in button.
   */
  iconLeft?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the right in button.
   */
  iconRight?: boolean
  /**
   * Used to enable/disable icons
   * Icons align in the center of the button.
   */
  icon?: boolean
  /**
   * Disables onPress option for button
   */
  disabled?: boolean
  active?: boolean
  inputButton?: boolean
  full?: boolean
  light?: boolean
  dark?: boolean
  /**
   * [android] colored ripple effect
   */
  androidRippleColor?: string
  /**
   * Segments
   */
  first?: boolean
  last?: boolean
  /**
   * Adds necessary padding when Text button defined in Left / Right of Header (iOS)
   */
  hasText?: boolean
  // Others
  activeOpacity?: any
  onlyCenter?: boolean
  facebook?: boolean
  twitter?: boolean
  svgLeft?: boolean
}
