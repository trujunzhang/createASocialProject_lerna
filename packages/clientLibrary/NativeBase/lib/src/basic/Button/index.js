import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback, StyleSheet } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { Text } from '@appBasic/Text';
import computeProps from '@appUtils/computeProps';
import { mapPropsToStyleNames } from '@app/tools';
class Button extends React.Component {
    getInitialStyle() {
        return {
            borderedBtn: {
                borderWidth: this.props.bordered ? 1 : undefined,
                borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
            }
        };
    }
    prepareRootProps() {
        const defaultProps = {
            style: this.getInitialStyle().borderedBtn
        };
        if (Array.isArray(this.props.style)) {
            const flattenedStyle = this.props.style.reduce((accumulator, currentValue) => accumulator.concat(currentValue), []);
            return computeProps({ ...this.props, style: flattenedStyle }, defaultProps);
        }
        return computeProps(this.props, defaultProps);
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const children = Platform.OS === 'ios'
            ? this.props.children
            : React.Children.map(this.props.children, (child) => child && child.type === Text
                ? React.cloneElement(child, {
                    uppercase: variables.btnUppercaseAndroidText,
                    ...child.props
                })
                : child);
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            Platform['Version'] < 21) {
            return (React.createElement(TouchableOpacity, Object.assign({}, this.prepareRootProps(), { ref: (c) => (this._root = c), activeOpacity: this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5 }), children));
        }
        else {
            if (this.props.rounded) {
                let buttonStyle = { ...this.prepareRootProps().style };
                let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex;
                return (React.createElement(View, { style: [
                        { maxHeight: buttonStyle.height },
                        buttonStyle,
                        { paddingTop: undefined, paddingBottom: undefined }
                    ] },
                    React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), background: this.props.androidRippleColor
                            ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                            : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true) }, this.prepareRootProps()),
                        React.createElement(View, { style: [
                                styles.childContainer,
                                {
                                    paddingTop: buttonStyle.paddingTop,
                                    paddingBottom: buttonStyle.paddingBottom,
                                    height: buttonStyle.height,
                                    flexGrow: buttonFlex
                                }
                            ] }, children))));
            }
            else {
                return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), onPress: this.props.onPress, background: this.props.transparent
                        ? TouchableNativeFeedback.Ripple('transparent')
                        : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false) }, this.prepareRootProps()),
                    React.createElement(View, Object.assign({}, this.prepareRootProps()), children)));
            }
        }
    }
}
Button.contextTypes = {
    theme: PropTypes.object
};
// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };
const styles = StyleSheet.create({
    childContainer: {
        flexShrink: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export { StyledButton as Button };
//# sourceMappingURL=index.js.map