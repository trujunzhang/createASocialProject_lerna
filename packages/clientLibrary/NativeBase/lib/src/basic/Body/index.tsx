import * as React from 'react'
import * as PropTypes from 'prop-types'
import { View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'
import { mapPropsToStyleNames } from '@app/tools'

import { BodyProps } from './Body'

class Body extends React.Component<BodyProps, any> {
  private _root: any
  render() {
    return <View name={'body'} ref={(c) => (this._root = c)} {...this.props} />
  }
}

// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }

const StyledBody = connectStyle('NativeBase.Body', {}, mapPropsToStyleNames)(Body)

export { StyledBody as Body }
