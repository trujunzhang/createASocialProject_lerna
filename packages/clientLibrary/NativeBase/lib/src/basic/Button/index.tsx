import * as React from 'react'
import * as PropTypes from 'prop-types'
import {
  TouchableOpacity,
  Platform,
  View,
  TouchableNativeFeedback,
  StyleSheet
} from 'react-primitives'
import { connectStyle } from 'native-base-shoutem-theme'
import { IThemeVariables } from '@app/tools' // typings
import { platformVariables as variable } from '@appVariables/theme/variables/platform' // variables // [variable]

import { Text } from '@appBasic/Text'
import computeProps from '@appUtils/computeProps'

import { mapPropsToStyleNames } from '@app/tools'

import { ButtonProps } from './Button'

class Button extends React.Component<ButtonProps, any> {
  static contextTypes = {
    theme: PropTypes.object
  }

  getInitialStyle() {
    return {
      borderedBtn: {
        borderWidth: this.props.bordered ? 1 : undefined,
        borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
      }
    }
  }

  // _root: React$Element<TouchableOpacity | TouchableNativeFeedback>;
  _root: any

  prepareRootProps() {
    const defaultProps = {
      style: this.getInitialStyle().borderedBtn
    }

    if (Array.isArray(this.props.style)) {
      const flattenedStyle = this.props.style.reduce(
        (accumulator: any, currentValue) => accumulator.concat(currentValue),
        []
      )
      return computeProps({ ...this.props, style: flattenedStyle }, defaultProps)
    }

    return computeProps(this.props, defaultProps)
  }
  render() {
    const variables: IThemeVariables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    const children =
      Platform.OS === 'ios'
        ? this.props.children
        : React.Children.map(this.props.children, (child: any) =>
            child && child.type === Text
              ? React.cloneElement(child, {
                  uppercase: variables.btnUppercaseAndroidText,
                  ...child.props
                })
              : child
          )
    if (
      Platform.OS === 'ios' ||
      Platform.OS === 'web' ||
      variables.androidRipple === false ||
      Platform['Version'] < 21
    ) {
      return (
        <TouchableOpacity
          {...this.prepareRootProps()}
          ref={(c) => (this._root = c)}
          activeOpacity={this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5}>
          {children}
        </TouchableOpacity>
      )
    } else {
      if (this.props.rounded) {
        let buttonStyle = { ...this.prepareRootProps().style }
        let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex
        return (
          <View
            style={[
              { maxHeight: buttonStyle.height },
              buttonStyle,
              { paddingTop: undefined, paddingBottom: undefined }
            ]}>
            <TouchableNativeFeedback
              ref={(c) => (this._root = c)}
              background={
                this.props.androidRippleColor
                  ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                  : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true)
              }
              {...this.prepareRootProps()}>
              <View
                style={[
                  styles.childContainer,
                  {
                    paddingTop: buttonStyle.paddingTop,
                    paddingBottom: buttonStyle.paddingBottom,
                    height: buttonStyle.height,
                    flexGrow: buttonFlex
                  }
                ]}>
                {children}
              </View>
            </TouchableNativeFeedback>
          </View>
        )
      } else {
        return (
          <TouchableNativeFeedback
            ref={(c) => (this._root = c)}
            onPress={this.props.onPress}
            background={
              this.props.transparent
                ? TouchableNativeFeedback.Ripple('transparent')
                : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false)
            }
            {...this.prepareRootProps()}>
            <View {...this.prepareRootProps()}>{children}</View>
          </TouchableNativeFeedback>
        )
      }
    }
  }
}

// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };

const styles = StyleSheet.create({
  childContainer: {
    flexShrink: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button)
export { StyledButton as Button }
