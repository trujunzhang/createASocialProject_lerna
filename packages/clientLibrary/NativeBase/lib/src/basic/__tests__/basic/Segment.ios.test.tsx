import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Container } from '../../Container'
import { Header } from '../../Header'
import { Left } from '../../Left'
import { Button } from '../../Button'
import { Icon } from '../../Icon'
import { Body } from '../../Body'
import { Right } from '../../Right'
import { Text } from '../../Text'
import { Segment } from '../../Segment'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//  return Platform;
// });

it('renders Segments inside Header', () => {
  const tree = renderer
    .create(
      <Container>
        <Header hasSegment>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Segment>
              <Button active={true} first>
                <Text>Puppies</Text>
              </Button>
              <Button last active={false}>
                <Text>Cubs</Text>
              </Button>
            </Segment>
          </Body>
          <Right>
            <Button transparent>
              <Icon name="search" />
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Segments outside Header', () => {
  const tree = renderer
    .create(
      <Container>
        <Header hasSegment />
        <Segment>
          <Button first active={true}>
            <Text>Puppies</Text>
          </Button>
          <Button active={false}>
            <Text>Kittens</Text>
          </Button>
          <Button last active={false}>
            <Text>Cubs</Text>
          </Button>
        </Segment>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
