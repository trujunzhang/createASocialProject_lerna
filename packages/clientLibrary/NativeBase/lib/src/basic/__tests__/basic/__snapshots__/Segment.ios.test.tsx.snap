// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Segments inside Header 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <view
            name="View"
            style={
              Object {
                "borderBottomLeftRadius": 2,
                "borderBottomRightRadius": 2,
                "borderTopLeftRadius": 2,
                "borderTopRightRadius": 2,
              }
            }
          >
            <view
              name="View"
              style={Object {}}
            >
              <text
                style={Object {}}
              >
                Puppies
              </text>
            </view>
          </view>
          <view
            name="View"
            style={
              Object {
                "borderBottomLeftRadius": 2,
                "borderBottomRightRadius": 2,
                "borderTopLeftRadius": 2,
                "borderTopRightRadius": 2,
              }
            }
          >
            <view
              name="View"
              style={Object {}}
            >
              <text
                style={Object {}}
              >
                Cubs
              </text>
            </view>
          </view>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-search
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Segments outside Header 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    />
  </view>
  <view
    name="View"
    style={Object {}}
  >
    <view
      name="View"
      style={
        Object {
          "borderBottomLeftRadius": 2,
          "borderBottomRightRadius": 2,
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        }
      }
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Puppies
        </text>
      </view>
    </view>
    <view
      name="View"
      style={
        Object {
          "borderBottomLeftRadius": 2,
          "borderBottomRightRadius": 2,
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        }
      }
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Kittens
        </text>
      </view>
    </view>
    <view
      name="View"
      style={
        Object {
          "borderBottomLeftRadius": 2,
          "borderBottomRightRadius": 2,
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        }
      }
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Cubs
        </text>
      </view>
    </view>
  </view>
</view>
`;
